exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityClass.kt

```diff
@@ -568,33 +568,88 @@ abstract class EntityClass<ID : Comparable<ID>, out T : Entity<ID>>(
         registerRefRule(column) { OptionalReferrers<ID, Entity<ID>, TargetID, Target, REF>(column, this, cache) }
 
     /**
-     * Returns a [ColumnWithTransform] delegate that transforms this stored [TColumn] value on every read.
+     * Returns a [EntityFieldWithTransform] delegate that transforms this stored [TColumn] value on every read.
+     *
+     * @param transformer An instance of [ColumnTransformer] to handle the transformations.
+     * @sample org.jetbrains.exposed.sql.tests.shared.entities.TransformationsTable
+     * @sample org.jetbrains.exposed.sql.tests.shared.entities.TransformationEntity
+     */
+    fun <TColumn : Any?, Target : Any?> Column<TColumn>.transform(
+        transformer: ColumnTransformer<TColumn, Target>
+    ): EntityFieldWithTransform<TColumn, Target> = EntityFieldWithTransform(this, transformer, false)
```

Comment:
With the introduction of `transform()` in DSL, I'm still not sure that I see the value of DAO's `transform()` being kept. They serve the same purpose (except for DAO's additional caching to avoid multiple expensive transformations) and they have the same name. At the moment, I can't think of any other example in the API that has separate DSL and DAO components that are identically named and also so close in functionality.

For example, using the DAO test file `ColumnWithTransformTest.kt` (of course without the new DSL+DAO unit tests):
If I cut & paste all the `transform` blocks from each entity class and place them in the respective table object instead, all tests still pass. So our current tests at least get the same result using the new functionality on the DSL table instead of the old functionality on the entity class.

I guess I just still don't have an answer to my original question: Why would/should a user choose `transform` on `Entity` instead of `transform` on `Table`?
If the answer is, just for potential memoisation, we could still enable that through the new DSL implementation by leaving the actual transformation logic in the core module only, deprecating the DAO `transform()`, & reducing the DAO `EntityFieldWithTransform` to only having the caching logic (& renaming function to something like `withTransformation()`).

Reply:
For me it looks the opposite way: I usually choose if I need Entity or Table and next stick with what I have. Let's discuss it anyway